src/components/SignUpForm/index.tsx

```diff
@@ -0,0 +1,318 @@
+import { useState, useCallback, SyntheticEvent } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  ID_REQUIRE_CHECK,
+  NICKNAME_REQUIRE_CHECK,
+  PW_REQUIRE_CHECK,
+  PW_VALID_CHECK,
+  PWCONFIRM_REQUIRE_CHECK,
+  PWCONFIRM_VALID_CHECK,
+  EMAIL_REQUIRE_CHECK,
+  EMAIL_VALID_CHECK,
+  MBTI_REQUIRE_CHECK,
+} from '../../constants/message';
+import { regEmail } from '../../constants/regEx';
+import LabelBasicInput from '../LabelBasicInput';
+import LabelBasicSelect from '../LabelBasicSelect';
+import Button from '../styled-components/Button';
+import './style.scss';
+
+// Mbti select option interface
+interface Mbti {
+  id: string;
+  name: string;
+}
+
+// Gender select option interface
+interface Gender {
+  id: string;
+  name: string;
+}
+
+const SignUpForm = () => {
+  const [loginId, setLoginId] = useState<string>('');
+  const [nickname, setNickname] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [passwordConfirm, setPasswordConfirm] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [mbti, setMbti] = useState<string>('1');
+  const [gender, setGender] = useState<string>('1');
```

Comment:
```suggestion
  const [state, setState] = useState<SignUpState>({
    loginId: '',
    nickname: '',
    password: '',
    passwordConfirm: '',
    email: '',
    mbti: '',
    gender: 'male',
  });
```
state하나로 관리할 수 있을 거 같아요.